---
title: "R_Assignment"
output: html_document
---
###BCB546X Fall 2019
##R Assignment
## David D. Ray

##Data Import

  #clone repository into R working directory
  #set working directory to UNIX_Assignment in order to load required data sets
Load Data
```{r}
library(tidyverse) #load tidyverse in order to use readr and dplyr
```
```{r}
fang_et_al_genotypes <- read_tsv("fang_et_al_genotypes.txt") #loads and renames data set
```
```{r}
snp_position <- read_tsv("snp_position.txt") #loads and renames data set
```


##Part I: Data Inspection of fang_et_al_genotypes
```{r}
file.info("fang_et_al_genotypes")
```
```{r}
file.info("snp_position")
```



```{r}
dim(fang_et_al_genotypes) #give the number of rows followed by the number of columns
```
```{r}
dim(snp_position) #give the number of rows followed by the number of columns
```


```{r}
colnames(fang_et_al_genotypes)   #col.names() #gives the names of columns

```
```{r}
colnames(snp_position) #col.names() #gives the names of columns
```
  
  
```{r}
row.names(fang_et_al_genotypes) #row.names() #gives the names of the row
```
```{r}
row.names(snp_position) #row.names() #gives the names of the row

```


##Data Processing
####snp_position.txt has the orientation we need that don't require transposition, we need only columns 1(SNP_ID), 2(Chromosome), and 4(Position)
```{r}
snp_ID_Chr_Pos <- snp_position[,-c(2,5:15)] #this trims columns 2,5, and everything past 5

```
  
####fang_et_al_genotype Data Processing
```{r}
maize_fang_et_al_genotypes <- filter(fang_et_al_genotypes, Group == "ZMMLR" | Group == "ZMMIL" | Group == "ZMMMR") #filter/sort groups
```
```{r}
teosinte_fang_et_al_genotypes <- filter(fang_et_al_genotypes, Group == "ZMPBA" | Group == "ZMPIL" | Group == "ZMPJA") #filter/sort groups
```
```{r}
maize_fang_et_al_genotypes <- maize_fang_et_al_genotypes[,-c(1,2,3)]
teosinte_fang_et_al_genotypes <- teosinte_fang_et_al_genotypes[,-c(1,2,3)]
 #remove columns 1:3 (sample_ID, JG_OTU, Groups)
```
```{r}
maize_fang_et_al_genotypes <- t (maize_fang_et_al_genotypes) #transpose newly created data sets and places SNP_ID in position 1,1
teosinte_fang_et_al_genotypes <- t (teosinte_fang_et_al_genotypes) #transpose newly created data sets and places SNP_ID in position 1,1
 
```
```{r}
  #Sort a data frame rows in ascending order (from low to high)
arrange(snp_ID_Chr_Pos, SNP_ID)
  #this syntax takes column SNP_ID from file snp_ID_Chr_Pos and arranges the values from low to high
``` 
  
##Combined data sets
```{r}
maize_fang_et_al_genotypes <- cbind(snp_position, maize_fang_et_al_genotypes)
  #syntax is we are combinding snp_ID_Chr_Pos with maize_fang_et_al_genotypes at the 1,1 position
teosinte_fang_et_al_genotypes <- cbind(snp_position, teosinte_fang_et_al_genotypes)
 #syntax is we are combinding snp_ID_Chr_Pos with teosinte_fang_et_al_genotypes at the 1,1 position
```

##Data Arrangment

  #For maize (Group = ZMMIL, ZMMLR, and ZMMMR in the third column of the fang_et_al_genotypes.txt file) we want 20 files in total:


10 files (1 for each chromosome) with SNPs ordered based on increasing position values and with missing data encoded by this symbol: ?
 
   for (i in 1:10) {
  maize <- filter(maize_fang_et_al_genotypes, Chromosome == i)
  maize <- rename(maize, "NA" | "", "?")
  maize <- arrange(maize, Position)}
    #this loops chromosomes 1:10 to first be filtered into a new dataset with the Chromosome column being ordered in increasing position, the new dataset is then edited to have have all missing data (NA) or blank values renamed with the "?" variable, this new data set is then arranged by the Position column

10 files (1 for each chromosome) with SNPs ordered based on decreasing position values and with missing data encoded by this symbol: -
```{r}
for (i in 1:10) {
  maize <- filter(maize_fang_et_al_genotypes, Chromosome == i)
  maize <- rename(maize, "?/?" , "-")
  maize <- arrange(maize, Position)}
    #this loops chromosomes 1:10 to first be filtered into a new dataset with the Chromosome column being ordered in increasing position, the new dataset is then edited to have have all missing data (NA) or blank values renamed with the "-" variable, this new data set is then arranged by the Position column
```
For teosinte (Group = ZMPBA, ZMPIL, and ZMPJA in the third column of the fang_et_al_genotypes.txt file) we want 20 files in total:

10 files (1 for each chromosome) with SNPs ordered based on increasing position values and with missing data encoded by this symbol: ?

for (i in 1:10) {
  teosinte <- filter(teosinte_fang_et_al_genotypes, Chromosome == i)
  teosinte <- rename(teosinte, "NA" | "", "?")
  teosinte <- arrange(teosinte, Position)}
    #this loops chromosomes 1:10 to first be filtered into a new dataset with the Chromosome column being ordered in increasing position, the new dataset is then edited to have have all missing data (NA) or blank values renamed with the "?" variable, this new data set is then arranged by the Position column

10 files (1 for each chromosome) with SNPs ordered based on decreasing position values and with missing data encoded by this symbol: -

teosinte <- filter(teosinte_fang_et_al_genotypes, Chromosome == i)
  teosinte <- rename(teosinte, "NA" | "", "-")
  teosinte <- arrange(teosinte, Position)}
    #this loops chromosomes 1:10 to first be filtered into a new dataset with the Chromosome column being ordered in increasing position, the new dataset is then edited to have have all missing data (NA) or blank values renamed with the "-" variable, this new data set is then arranged by the Position column



A total of 40 files will therefore be produced.
  
  



````
```{r}
```

















